core/src/main/java/org/elasticsearch/action/admin/cluster/repositories/put/TransportPutRepositoryAction.java

```diff
@@ -44,8 +46,8 @@
     @Inject
     public TransportPutRepositoryAction(Settings settings, TransportService transportService, ClusterService clusterService,
                                         RepositoriesService repositoriesService, ThreadPool threadPool, ActionFilters actionFilters,
-                                        IndexNameExpressionResolver indexNameExpressionResolver) {
-        super(settings, PutRepositoryAction.NAME, transportService, clusterService, threadPool, actionFilters, indexNameExpressionResolver, PutRepositoryRequest::new);
+                                        IndexNameExpressionResolver indexNameExpressionResolver, TaskManager taskManager) {
+        super(settings, PutRepositoryAction.NAME, transportService, clusterService, threadPool, actionFilters, indexNameExpressionResolver, PutRepositoryRequest::new, taskManager);
```

Comment:
honestly I dont' think this is the way to go. I looked at like 10 classses and non of them uses the `TaskManager` nor the `Task`. I think What we should do is we should expose `TaskManager` via `ClusterService` and make the `ClusterService` instantiate `TaskManager`, that safes all the ctor changes.

for the `masterOperation` add  this to `TransportMasterNodeAction`:

```Java
protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<Response> listener) throws Exception {
  masterOperation(request, state, listener); // just delegate if we need the task we can override
} 
```